torch/tensor.py

```diff
@@ -423,9 +423,9 @@ def __contains__(self, element):
             element (Tensor or scalar): element to be checked
                 for presence in current tensor"
         """
-        if isinstance(element, (torch.Tensor, Number)):
-            return (element == self).any().item()
-        return NotImplemented
+        if not isinstance(element, (torch.Tensor, Number)):
+            raise NotImplementedError
```

Comment:
Based on the documentation and @gchanan's comments on the original issue,
- https://docs.python.org/3.7/library/exceptions.html#NotImplementedError
- https://docs.python.org/3.7/library/constants.html#NotImplemented

it sounds like we should not be raising a NotImplementedError. The exact reasoning is that the documentation says:
- "It should not be used to indicate that an operator or method is not meant to be supported at all – in that case either leave the operator / method undefined or, if a subclass, set it to None."
- "In user defined base classes, abstract methods should raise this exception when they require derived classes to override the method, or while the class is being developed to indicate that the real implementation still needs to be added."

None of those two cases really apply, but let me know what you think.

Could you instead raise a RuntimeError with an error message of something along the lines of "`Tensor.__contains__` only supports scalars, but you passed in a {}. Please file a bug if you believe this message to be incorrect."

Reply:
That's better! Learn a lot from your comment.